internal/pkg/deploy/cloudformation/stack/validate.go

```diff
@@ -100,6 +114,154 @@ func validateSidecarMountPoints(in []manifest.SidecarMountPoint) error {
 	return nil
 }
 
+func validateSidecarDependsOn(in manifest.SidecarConfig, sidecarName string, s convertSidecarOpts) error {
+	if in.DependsOn == nil {
+		return nil
+	}
+
+	for name, status := range in.DependsOn {
+		// status must be one of < start | complete | success >
+		if !isValidStatus(status) {
+			return errInvalidDependsOnStatus
+		}
+		// essential containers must have 'start' as a status
+		if name == s.workloadName || s.sidecarConfig[name].Essential == nil || aws.BoolValue(s.sidecarConfig[name].Essential) {
+			if status != "start" {
+				return errEssentialContainerStatus
+			}
+		}
+	}
+
+	return nil
+}
+
+func validateNoCircularDependencies(s convertSidecarOpts) error {
+	used := make(map[string]bool)
+	path := make(map[string]bool)
+	dependencies, err := buildDependencyGraph(s)
+
+	// don't let nonexistent containers pass
+	if err != nil {
+		return err
+	}
+
+	// don't let circular dependencies happen
```

Comment:
```suggestion
	// don't let circular dependencies happen
```
I think one thumb of rule is if the code itself is explicit enough then we don't need additional comment (we should avoid relying too much on the comment to increase the code readability)

Reply:
Adjusted comments.